txjuju/tests/test_api.py

```diff
@@ -934,6 +934,42 @@ def test_modelInfo_bad_result(self):
         self.assertEqual("malformed result {}", err.value.error)
         self.assertEqual("", err.value.code)
 
+    def test_setModelConfig_good_result(self):
+        """
+        The setModelConfig method sends an 'ModelConfig' request and
+        returns a deferred that will callback with a ModelInfo instance.
+        """
+        deferred = self.client.setModelConfig(
+            "automatically-retry-hooks", False)
+        params = {"config": {"automatically-retry-hooks": False}}
+        self.assertEqual("ModelConfig", self.backend.lastType)
+        self.assertEqual("ModelSet", self.backend.lastRequest)
+        self.assertEqual(1, self.backend.lastVersion)
+        self.assertEqual(params, self.backend.lastParams)
+        self.backend.response({})
+        self.assertIsNone(self.successResultOf(deferred))
```

Comment:
While the API returns an empty dict {} in response, our _parseSetModelConfigResult essentially discards anything other than error messages and returns None.